drivers: can: Set a initial state to the can device before HAL_CAN_Init

sritej20 opened the pull request:
As part of HAL_CAN_Init we check the initial state of the can handle.
Setting it to HAL_CAN_STATE_RESET as an initial state to start the
Init properly.

Also included a bunch of changes to satisfy uncrustify on the file.

Resolves: #8416
Coverity-CID: 186580

Signed-off-by: Sritej Kanakadandi Venkata Rama <sritej.kvr@gmail.com>

sritej20 commented on drivers/can/stm32_can.c:
```diff
@@ -205,6 +205,8 @@ int can_stm32_runtime_configure(struct device *dev, enum can_mode mode,
 	hcan.Init.BS2  = bs2;
 	hcan.Init.Prescaler = prescaler;
 
+	hcan.State = HAL_CAN_STATE_RESET;
```
Actual change that i made to initialize the state of the Can handle before init ( since Init checks the state before initializing)

alexanderwachter commented:
@sritej20 could you split this in two commits?
One for the actual bug and one for the refactoring.

sritej20 committed:
drivers: can: Set a initial state to the can device before HAL_CAN_Init

As part of HAL_CAN_Init we check the initial state of the can handle.
Setting it to HAL_CAN_STATE_RESET as an initial state to start the
Init properly.

Resolves: #8416
Coverity-CID: 186580

Signed-off-by: Sritej Kanakadandi Venkata Rama <sritej.kvr@gmail.com>

sritej20 commented:
@alexanderwachter Updated PR to only have the change im making. Will open a new PR with the rest of the refactoring later.

alexanderwachter reviewed (approved):
Looks good. Thanks for fixing this issue!

sritej20 commented:
Thanks for the approval @alexanderwachter
